woocommerce/woocommerce-android
# Using site timezone in orders list

Part of: #9887 

### Why
This PR continues the work of using the store's time zone on the app.  As discussed here pe5pgL-4bG-p2, besides the stats, we also want to use the store time zone for the order date:
> we will use the store time zone for everything order-related (orders, order notes, etc.)

### Description
This PR modifies the dates used in the orders list to match the store's time zone. It does that by moving the logic to create the date description to the data source and using the new date util functions to parse the dates using the store's time zone.

### Testing instructions
To test these changes, you must set your device to a different time zone than your store. I used a store in the US and set the device time zone to Australia. This will create a time zone difference of one day (Thursday in the store should be Friday on the device)

TC1
1. Open the store and place a new order (on the web)
2. Open the app 
3. Open the orders list
4. Check that the new order appears under the Today section and that the date matches the store's time zone 

TC2
1. Open the orders list screen
2. Tap on add (+) button
3. Create a new order
4. Go back to the orders list and refresh the screen
5. Check that the new order appears under the Today section and that the date matches the store's time zone 

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/18119390/c50f2482-8307-4df1-bfe3-25ed2265ed7f
